lib/primitives/array-element.js

```diff
@@ -107,8 +111,47 @@ module.exports = function(BaseElement, registry) {
       return newArray;
     },
 
+    /*
+     * This is a reduce function specifically for Minim arrays and objects. It
+     * allows for returning normal values or Minim instances, so it converts any
+     * primitives on each step.
+     */
+    reduce: function(fn, givenMemo) {
+      var startIndex;
+      var memo;
+
+      // Allows for defining a starting value of the reduce
+      if (!_.isUndefined(givenMemo)) {
+        startIndex = 0;
+        memo = registry.toElement(givenMemo);
+      } else {
+        startIndex = 1;
+        // Object Element content items are member elements. Because of this,
+        // the memo should start out as the member value rather than the
+        // actual member itself.
+        memo = this.primitive() === 'object' ? this.first().value : this.first();
+      }
+
+      // Sending each function call to the registry allows for passing Minim
+      // instances through the function return. This means you can return
+      // primitive values or return Minim instances and reduce will still work.
+      for (var i = startIndex; i < this.length; i++) {
+        var item = this.content[i];
+
+        if (this.primitive() === 'object') {
+          memo = registry.toElement(fn(memo, item.value, item.key, item));
+        } else {
+          memo = registry.toElement(fn(memo, item));
```

Comment:
We might want to use `fn(memo, item, i, this.content)` to mimic the [`Array.prototype.reduce`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce)